src/examples/pwm/pwm.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"machine"
+	"time"
+)
+
+// This example assumes that an RGB LED is connected to pins 3, 5 and 6 on an Arduino.
+// Change the values below to use different pins.
+const (
+	redPin   = 3
+	greenPin = 5
+	bluePin  = 6
+)
+
+// cycleColor is just a placeholder until math/rand or some equivalent is working.
+func cycleColor(color uint16) uint16 {
+	if color < 10 {
+		return color + 1
+	}
+	if color > 30 && color < 200 {
+		return color + 10
+	}
+	if color > 230 {
+		return 0
+	}
```

Comment:
I don't think this function works correctly: it causes all colors to be in the range 0..10. Is that the intention?
Otherwise, maybe it should be like this? It slowly builds up from 0 to 10 and then jumps in bigger steps until 200 to start again at 0.
```go
func cycleColor(color uint8) uint8 {
        if color < 10 {
                return color + 1
        } else if color < 200 {
                return color + 10
        } else {
                return 0
        }
}
```
Discovered while testing the code.

Reply:
It is a bit ugly but it does test values greater than 10. This can be visually determined when looking at the LED.

Actually I was going to replace with a specific list of RGB colors to make it easier to spot check, but it does send several RGB colors to exercise various PWM duty cycle values, which was the main point.

Reply:
I printed the numbers and for me it certainly never reaches beyond 10 and the LED stays rather dim. The LED gets a whole lot brighter with the above code.

Also, consider what happens when `cycleColor()` gets the value 10:

```go
	if color < 10 { // if not taken: 10 isn't smaller than 10
		return color + 1
	}
	if color > 30 && color < 200 { // if not taken: 10 isn't bigger than 30
		return color + 10
	}
	if color > 230 { // if not taken: 10 isn't bigger than 230
		return 0
	}
	// cycleColor(10) will return 0
	return 0
```

The same result can be obtained with input values 20 and 30 (for `gc` and `bc`) - these also return 0 restarting from the beginning.

Reply:
Sure, sounds good. Please change to your proposes function, thanks.